priv/static/phoenix_live_reload.js

```diff
@@ -53,6 +55,27 @@ socket.connect();
 var chan = socket.channel('phoenix:live_reload', {})
 chan.on('assets_change', function(msg) {
   var reloadStrategy = reloadStrategies[msg.asset_type] || reloadStrategies.page;
+
+  if (restoreScrollOnReload && reloadStrategy === pageStrategy) {
+    sessionStorage.setItem(SESSION_STORAGE_SCROLL_Y_KEY, window[targetWindow].scrollY);
+  }
+
   setTimeout(function(){ reloadStrategy(chan); }, interval);
 });
-chan.join();
+
+var optionallyRestoreScroll = function() {
+  if (restoreScrollOnReload) {
+    const scrollYSerialized = sessionStorage.getItem(SESSION_STORAGE_SCROLL_Y_KEY)
+
+    if (scrollYSerialized) {
+      const scrollY = parseInt(scrollYSerialized, 10)
+
+      window[targetWindow].scrollTo(0, scrollY);
+      sessionStorage.removeItem(SESSION_STORAGE_SCROLL_Y_KEY);
+    }
+  }
+}
+
+chan
+  .join()
+  .receive('ok', optionallyRestoreScroll);
```

Comment:
`optionallyRestoreScroll` restores a position at most once (it clears the sessionStorage value), so we should be protected from stale values getting referenced in the case of channel rejoins.

Reply:
@chrismccord I should also note that the new `window.scrollTo/2` invocation I'm adding on 'join' should not conflict with [the `window.scrollTo(0, 0)` invocation that occurs in LiveView applications after a patch](https://github.com/phoenixframework/phoenix_live_view/pull/2173), as the new invocation only occurs on channel join (on initial browser load/bootstrap).